src/libraries/System.Private.Xml/src/System/Xml/Schema/XsdDateTime.cs

```diff
@@ -396,6 +396,7 @@ InternalKind switch
             {
                 case DateTimeTypeCode.GMonth:
                 case DateTimeTypeCode.GDay:
+                    // codeql[cs/leap-year/unsafe-date-construction-from-two-elements] - The XML specification does not explicitly define this behavior for parsing in a non-leap year. We intentionally throw here. Altering this behavior to be more resilient, producing dates like 2/28 or 3/1, could introduce unintended consequences and may not be desirable for user.
                     result = new DateTime(DateTime.Now.Year, xdt.Month, xdt.Day);
```

Comment:
It can throw, the parsing code is just use leap year during the parsing which will make 2/29 valid parsing. 

result = new DateTime(DateTime.Now.Year, xdt.Month, xdt.Day);

Is using `DateTime.Now.Year` which is possible to be non-leap year. This will throw .

Reply:
But it's not passing `xdt.Year` it's passing `DateTime.Now.Year`

Reply:
I think you are right @KalleOlaviNiemitalo. We should fix the comment. Are you interested to submit a PR?